micro-optimizations + no ellipses

Resuscitating opened the pull request.

Commit:
Merge pull request #1 from thatbirdguythatuknownot/main

keeping my fork up to date

Commit:
Merge pull request #2 from thatbirdguythatuknownot/main

make up to date with main fork

Commit:
micro-optimize self.cache default entries

micro-optimize self.cache default entries, and remove use of ellipses in _default_object_repr_pair_W and _default_object_repr_pair, since it says only dunders.
I'm working on some bigger stuff than micro-optimizations, but since it's not done yet I figured I'd put this in now instead of waiting till later.

thatbirdguythatuknownot commented on obfuscator.py:
```diff
@@ -109,8 +109,8 @@ class Obfuscator:
-            0: "__name__.__ne__(__name__)",
-            1: "__name__.__eq__(__name__)"
+            0: "().__len__()",
+            1: "((),).__len__()",
```
One of the general rules for this obfuscator (which I haven't written officially yet) is that there should be **no** constants (ellipses included, removed in the current local updated version of `obfuscator.py`). `()` and `((),)` are constants, so that won't be accepted.

thatbirdguythatuknownot replied:
I might update the Github file later with the latest local version currently stored in my computer.
What I do know is that `None` is `__name__.__getstate__()`, `1` is `__name__.__eq__(__name__).__pos__()`, `0` is `__name__.__class__().__len__()`, and an empty string is now `__name__.__class__()`.

Commit:
comply with no constants rule

comply with no constants rule by removing tuples
add rules docstring at the top of the module
change stringified functions to comments
remove redundant check in UnparseObfuscate.traverse

Commit:
minor formatting

Commit:
use __name__.__getstate__

use __name__.__getstate__ instead of __annotations__.__lt__(__name__).__doc__

Commit:
Update obfuscator.py

use __name__.__class__.__base__ instead of __name__.__class__.__class__.__base__ to get object, ~1.4x faster on my machine using timeit
micro-optimizations :D

ghost commented:
fixed the usage of constants, micro-optimized some stuff, wrote some rules at the top of the module

thatbirdguythatuknownot commented:
> fixed the usage of constants, micro-optimized some stuff, wrote some rules at the top of the module

Thanks, I'll have some of the changes written to the local version and then update the Github file.
